src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/PackageTask.groovy

```diff
@@ -21,18 +21,44 @@ class PackageTask extends AbstractTask {
 
     @TaskAction
     void packageServer() {
+	
         def params = buildLibertyMap(project);
-        if (project.liberty.packageLiberty.archive != null && project.liberty.packageLiberty.archive.length() != 0) {
-            params.put('archive', new File(project.liberty.packageLiberty.archive))
-        }
+		def fileType = getPackageFileType(project.liberty.packageLiberty.include)
+		final archive = project.liberty.packageLiberty.archive
+		
+        if (archive != null && archive.length() != 0) {
+			def packageFile = new File(project.getProjectDir(), archive)
+			
+			if (!archive.endsWith(".zip") && !archive.endsWith(".jar")) {
```

Comment:
You should honor the ${archive} setting in build.gradle.  If ${archive} exists and it is a directory, you will create the archive file based on the default filename and file type logic under this directory. If it is a file or doesn't exist, it will be the name for the archive file.  You can reference to liberty-maven-plugin implementation.